app/models/user.rb

```diff
@@ -329,12 +330,32 @@ def send_reset_password_instructions
     super
   end
 
-  def reset_password!(new_password, new_password_confirmation)
+  def reset_password(new_password, new_password_confirmation)
     return false if encrypted_password.blank?
 
     super
   end
 
+  def reset_password!
+    # First, change password to something random, invalidate the remember-me token,
+    # and deactivate all sessions
+    transaction do
+      update(remember_token: nil, remember_created_at: nil, password: SecureRandom.hex)
```

Comment:
Hm, isn't there a cleaner way to disable password login altogether? Shouldn't matter much in practice though.

Reply:
I don't want to disable password login, just make it impossible to login with the previously known password

Reply:
I don't understand. Don't you want it to make it impossible to login with any password until the password reset link is clicked by the user? Also, this triggers a password changed e-mail, which may be very confusing to the user.